job_controller/job_controller.go

```diff
@@ -60,6 +73,42 @@ type ControllerInterface interface {
 
 	// Returns the Job from API server
 	GetJobFromAPIClient(namespace, name string) (metav1.Object, error)
+
+	// Returns the default container in pod. e.g. tensorflow
+	GetDefaultContainerName() string
+
+	// Returns the default container port name. e.g. tfjob-port
+	GetDefaultContainerPortName() string
+
+	// DeleteJob deletes the job
+	DeleteJob(job interface{}) error
+
+	// UpdateJobStatus updates the job status objects with new job conditions based on replica statuses
+	// e.g. Check whether worker 0 is exited without error and set job status
+	UpdateJobStatus(job interface{}, replicas map[commonv1.ReplicaType]*commonv1.ReplicaSpec,
+		jobStatus *commonv1.JobStatus, restart bool) error
```

Comment:
I'm a little confused with `jobStatus` passed inhere, is this a new job status or just the inside attribution of `job` ?

Reply:
It's the inner field of the job.  we could get the fields by typecasting the job but that will assume the XXXJob must embed the common.JobStatus which may or may not be true. XXXJob could have its own XXXJobStatus which embeds common.JobStatus.